docs/control-plane-architecture.adoc

```diff
@@ -0,0 +1,252 @@
+
+IMPORTANT: The API should support arbitrarily large systems.  The system itself should be clear about its target scale and avoid catastrophic degradation due to users consuming too many resources.
+
+**Security.**  Older versions of <<rfd6>> discussed control plane security in great detail.  That content needs to be extracted from the history and probably put here.
+
+**Supportability and debuggability.**  Effective customer support includes rapidly diagnosing issues and releasing fixes with low-risk updates.  To achieve this, all the software in the system, including the control plane, must be built with supportability in mind, which means being able to collect enough information about failures to diagnose them from their first occurrence in the field as much as possible and being able to update software with low risk to the system.  Details will be covered in an RFD to-be-named-later.
+
+== Parts of the control plane
+
+=== Crash course on hardware architecture
+
+For our purposes, an Oxide rack comprises three types of boards (systems):
+
+* Up to 32 compute **sleds** (servers).  These are sometimes called **Gimlets**, though "Gimlet" technically refers to a particular hardware generation.  Within the sled, the **host system** is the x86 box we generally think of as "the server".
+* 1 or 2 **switches**, each attached via PCIe to one of the 32 compute sleds.  (The switches are _also_ connected to each of the 32 sleds for networking.  This PCIe connection we're talking about is for control of the switch itself, which is only done by one sled.)  The chassis that house the switches are sometimes called **Sidecars**, though "Sidecar" technically refers to a particular hardware generation.  Sleds that are attached to switches are often called **Scrimlets** (which is a little unfortunate since the name obviously comes from "Gimlet", but it might not be a Gimlet (since Gimlet refers to a specific hardware generation)).
+* 1-2 power shelves, each with a **Power Shelf Controller (PSC)** that provides basic monitoring and control for the rectifiers that make up the power shelf.
+
+Each type of system (Gimlet, Sidecar, and PSC) contain a **service processor** (SP) that's responsible basic monitoring and control, typically including power control and thermal management.
+
+<<rfd210>> discusses service processors in more detail.
+
+=== Components that run alongside specific hardware
+
+.Overview of the control plane
+image::architecture-rev-2024-08-01.svg[Control Plane Architecture]
```

Comment:
In the diagram, for the boxes where there are 3 copies (CockroachDB, Nexus, Oximeter), are there actually 3 instances in reality?

Reply:
No, it's generally N.  This is elaborated in the text (e.g., table 2).